### ISSUE TITLE
Lsp not working

### ISSUE THREAD

**Issue by alvarogaiotti** [iamcco/coc-flutter#171]
**Describe the bug**
The lsp is not working with flutter version 3.9.0-1.0.pre.92.

**To Reproduce**
1. Open a .dart file in a flutter project having installed the specified flutter version
2. Try any lsp feature (autocompletion, jumpToDefinition, etc.)
3. Lsp is not working, the rest is fine

**Expected behavior**
Lsp working the standard way, providing completion and code actions etc.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Vim or Neovim: Neovim 
 - (Neo)vim version: 0.8.3

**Output channel:**

```
[dev-command]: register dev command
[sdk]: No local fvm sdk
[sdk]: which flutter command => C:\Users\gttlv\bin\flutter\bin\flutter.BAT
[sdk]: C:\Users\gttlv\bin\flutter\version => 3.9.0-1.0.pre.92
[sdk]: searchPaths: ~/snap/flutter/common/flutter
[sdk]: flutter command path => C:\Users\gttlv\bin\flutter\bin\flutter.bat
[sdk]: dart sdk home => C:\Users\gttlv\bin\flutter\bin\cache\dart-sdk
[sdk]: analyzer path => C:\Users\gttlv\bin\flutter\bin\cache\dart-sdk\bin\snapshots\analysis_server.dart.snapshot
[sdk]: dart command path => C:\Users\gttlv\bin\flutter\bin\cache\dart-sdk\bin\dart.exe
[fs]: executing command C:\Users\gttlv\bin\flutter\bin\cache\dart-sdk\bin\dart.exe --version
[sdk]: dart version: v3.0.0
[lsp-server]: rightVersion false
[lsp-server]: analysis server ready!
[lsp-closing-labels]: register closing labels
[daemon]: got message: {
  "event": "daemon.connected",
  "params": {
    "version": "0.6.1",
    "pid": 2140
  }
}
[daemon]: got message: {
  "event": "daemon.logMessage",
  "params": {
    "level": "status",
    "message": "Starting device daemon..."
  }
}
[daemon]: got message: {
  "id": 1
}
[daemon]: Message without event
[daemon]: got message: {
  "event": "device.added",
  "params": {
    "id": "windows",
    "name": "Windows",
    "platform": "windows-x64",
    "emulator": false,
    "category": "desktop",
    "platformType": "windows",
    "ephemeral": false,
    "emulatorId": null,
    "sdk": "Microsoft Windows [Versione 10.0.19044.2604]",
    "capabilities": {
      "hotReload": true,
      "hotRestart": true,
      "screenshot": false,
      "fastStart": false,
      "flutterExit": true,
      "hardwareRendering": true,
      "startPaused": true
    }
  }
}
[daemon]: New device: Windows
[daemon]: got message: {
  "event": "device.added",
  "params": {
    "id": "chrome",
    "name": "Chrome",
    "platform": "web-javascript",
    "emulator": false,
    "category": "web",
    "platformType": "web",
    "ephemeral": false,
    "emulatorId": null,
    "sdk": "Google Chrome 110.0.5481.178",
    "capabilities": {
      "hotReload": true,
      "hotRestart": true,
      "screenshot": false,
      "fastStart": false,
      "flutterExit": false,
      "hardwareRendering": true,
      "startPaused": true
    }
  }
}
[daemon]: New device: Chrome
[daemon]: got message: {
  "event": "device.added",
  "params": {
    "id": "edge",
    "name": "Edge",
    "platform": "web-javascript",
    "emulator": false,
    "category": "web",
    "platformType": "web",
    "ephemeral": false,
    "emulatorId": null,
    "sdk": "Microsoft Edge 110.0.1587.50",
    "capabilities": {
      "hotReload": true,
      "hotRestart": true,
      "screenshot": false,
      "fastStart": false,
      "flutterExit": false,
      "hardwareRendering": true,
      "startPaused": true
    }
  }
}
[daemon]: New device: Edge
```

**Comment by alvarogaiotti** [comment]
The error seems to be in src/lib/sdk.ts, line 91:
```typescript
return v.every((n, idx) => n >= version[idx]);
```
after forking the repo, changing the line to:
```typescript
return (v.every((n, idx) => n >= version[idx]) || v[0] > version[0]);
```
and installing the extension with vim-plug everything seems to work fine.

**Comment by resultanyildizi** [comment]
> and installing the extension with vim-plug and testing it `[lsp-server]: rightVersion` is true and everything seems to work fine.
Could you describe how to install the forked extension with vim-plug?

**Comment by Kavantix** [comment]
That version check is indeed incorrect, thanks for debugging. I’ll fix the comparison.

**Comment by Kavantix** [comment]
@phferreira @alvarogaiotti can you check if this is fixed in v1.9.10?

**Comment by Kavantix** [comment]
Actually this one was about dart 3, that should work in v1.9.10

@phferreira can you respond on the other ticket if your issues is also resolved
